components/admin/orgs/OrganisationControlForm.tsx

```diff
@@ -1,93 +1,77 @@
 import AdminTable, { AdminTableColumnProps } from '../AdminTable'
 
-function OrganisationControlForm() {
+type OrganisationControlFormProps = {
+  users: any[]
+  orgs: OrganisationWithIGHead[]
+  categories: any
+  mutateUsers: KeyedMutator<User[]>
+  mutateOrgs: KeyedMutator<OrganisationWithIGHead[]>
+}
+
+function OrganisationControlForm({
+  users,
+  orgs,
+  categories,
+  mutateOrgs,
+  mutateUsers,
+}: OrganisationControlFormProps) {
   const { isOpen, onOpen, onClose } = useDisclosure()
   const [auth] = useUserInfo()
   const toast = useToast()
-  const [initialValues, setInitialValues] = useState(defaultValues)
-
-  // const {
-  //   data: users,
-  //   error: errorUsers,
-  //   isLoading: isLoadingUsers,
-  //   mutate: mutateUsers,
-  // } = useSWR<User[], string[]>(
-  //   auth?.token ? [process.env.NEXT_PUBLIC_BACKEND_URL + 'users', auth.token] : null,
-  //   getFromUrlStringAndParseJsonWithAuth,
-  // )
-
-  const {
-    data: orgs,
-    error: errorOrgs,
-    isLoading: isLoadingOrgs,
-    mutate: mutateOrgs,
-  } = useSWR<OrganisationWithIGHead[], string[]>(
-    [process.env.NEXT_PUBLIC_BACKEND_URL, 'orgs'],
-    getFromUrlArrayAndParseJson,
-  )
-  const {
-    data: allOrgCategories,
-    error: errorOrgCategories,
-    isLoading: isLoadingOrgCategories,
-  } = useSWRImmutable<{ [key: string]: string }, string>(
-    process.env.NEXT_PUBLIC_BACKEND_URL + 'orgs/categories',
-    getFromUrlStringAndParseJson,
-  )
+  const [initialValues, setInitialValues] = useState<OrganisationForm>(defaultValues)
 
-  if (!isUserLoggedIn(auth) || auth === null) {
-    return <Box>Please log in first!</Box>
-  }
-
-  if (isLoadingOrgCategories || isLoadingOrgs) {
-    return <Box>Fetching data! Spinner</Box>
-  }
-
-  if (errorOrgCategories || errorOrgs) {
-    throw new Error("Could not fetch organisations' data from the backend")
+  if (auth === null) {
+    // should not occur as already checked in parent component
+    return <Box>Authentication Error</Box>
   }
 
   const onSubmit = async (
-    values: typeof initialValues,
+    values: OrganisationForm,
     { setSubmitting }: { setSubmitting: (isSubmitting: boolean) => void },
   ) => {
+    // need to parse igHead to number type as HTML option by default will parse the field as string
+    const parsedValues = { ...values, igHead: Number(values.igHead) }
     const { responseJson, responseStatus } = await makeFetchToUrlWithAuth(
-      process.env.NEXT_PUBLIC_BACKEND_URL + 'org',
+      process.env.NEXT_PUBLIC_BACKEND_URL + `org/${parsedValues.id}`,
       auth.token,
-      'POST',
-      JSON.stringify(values),
+      'PUT',
+      JSON.stringify(parsedValues),
     )
 
     if (responseStatus === 200) {
-      toast(makeSuccessOrgToast())
+      toast(
+        makeSuccessToast(
+          parsedValues.igHead === -1 ? `Org created successfully!` : `Org edited successfully`,
+        ),
+      )
       mutateOrgs()
+      mutateUsers()
```

Comment:
Note for me - this is an expedient hack to simply refetch all the data again, couldn't figure out why the mutate call to the API wouldn't reload useSWR()'s cached data. To fix if DO's limits are hit

Reply:
This might be tricky to resolve, I may just kick the can down the road for this and create an issue for another time